provider/datagen/src/driver.rs

```diff
@@ -972,36 +948,27 @@ fn select_locales_for_key(
         };
     }
 
-    if key == icu_segmenter::provider::DictionaryForWordOnlyAutoV1Marker::KEY
-        || key == icu_segmenter::provider::DictionaryForWordLineExtendedV1Marker::KEY
-    {
+    if key.path().get().starts_with("segmenter/dictionary/") {
```

Comment:
Question: why this change? Are we trying to make the driver not depend on the metacrate?

Except _maybe_ for blob-to-blob, I think we always need the data struct impls, because we need to either serialize (baked-to-blob), deserialize (blob-to-baked), or downcast (baked-to-baked since we need the ExportMarker intermediate). I think Dart is doing blob-to-baked, right? So it needs the data structs.

Even in blob-to-blob, you might need the data struct impls in case the formats are different (e.g. transform from Postcard to JSON). So, the case where you can eliminate the metacrate dependency is very narrow.

In that very narrow case, which is basically just postcard-to-postcard or json-to-json, I think we might need to modify ExportMarker, which isn't currently designed to be very modular:

```rust
trait ExportableDataPayload {
    fn bake_yoke(&self, env: &CrateEnv) -> TokenStream;
    fn serialize_yoke(
        &self,
        serializer: &mut dyn erased_serde::Serializer,
    ) -> Result<(), DataError>;
    fn as_any(&self) -> &dyn Any;
    fn eq_dyn(&self, other: &dyn ExportableDataPayload) -> bool;
}
```

I guess you could just `panic!` in the functions that you can't implement. `eq_dyn` should work because you can compare byte buffers. We would however need to figure out how to make `serialize_yoke` pass through buffers.

In any case, I think this is very much out of scope of this PR.

Reply:
Blob-to-blob won't need it, and blob-to-blob is the Dart case. The main point of removing the meta crate dependecies is because if we end up having three crates that depend on the meta crate, it gets really messy around features and experimental.

Reply:
Do you agree that the driver, or more specifically the BakedExporter and BlobExporter, need the metacrate dependency as at least an optional dependency in order to work with all source providers, and that the only case when the metacrate could theoretically be turned off (pending refactoring) would be postcard-to-postcard and json-to-json?

Reply:
BlobExporter does not need the meta crate, but BakedExporter does.

Reply:
- @sffc - I'm concerned about going from something that is type-safe to something that is not type-safe
- @robertbastian - They are constants, and I think it's safer to check for the `collator/` prefix that contains all collator keys, than to list the keys
- @sffc - I guess we could switch this to using `ResourceKeyMetadata` at some point
- @robertbastian - also if we add per-component features to the `DatagenProvider`, this shouldn't pull in collator and segmenter just to have special logic for their keys (and I don't want collator/segmenter features on the driver)